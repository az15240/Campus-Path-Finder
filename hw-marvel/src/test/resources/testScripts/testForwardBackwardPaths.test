# Tests data set from UWProfessorCustom.csv. Adapted from https://edstem.org/us/courses/32028/discussion/2601400.
# This file tests the forward and backward paths between two nodes. They might not necessarily be the
# opposite orders because we only care about the lexicographically least path.

LoadGraph g UWProfessorsCustom.csv

ListNodes g

FindPath g Schafer Grossman

FindPath g Grossman Schafer

FindPath g A.Rao Thachuk

FindPath g Thachuk A.Rao
