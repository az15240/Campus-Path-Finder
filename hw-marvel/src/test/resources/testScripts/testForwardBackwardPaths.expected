# Tests data set from UWProfessorCustom.csv. Adapted from https://edstem.org/us/courses/32028/discussion/2601400.
# This file tests the forward and backward paths between two nodes. They might not necessarily be the
# opposite orders because we only care about the lexicographically least path.

loaded graph g

g contains: A.Rao Beame Grossman Hsia J.Lee Karlin Kohlbrenner Kohno Leang Maas OveisGharan Perkins R.Lin Ri.Anderson Roesner Ru.Anderson Schafer Tang Tessaro Thachuk Tsun Weber Wilcox Y.Lee Zatloukal

path from Schafer to Grossman:
Schafer to Ru.Anderson via 391
Ru.Anderson to Hsia via 351
Hsia to Perkins via 333
Perkins to Wilcox via 331
Wilcox to Grossman via 341

path from Grossman to Schafer:
Grossman to Wilcox via 341
Wilcox to Perkins via 331
Perkins to Hsia via 333
Hsia to Ru.Anderson via 351
Ru.Anderson to Schafer via 391

path from A.Rao to Thachuk:
A.Rao to Beame via 431
Beame to Zatloukal via 311
Zatloukal to Perkins via 331
Perkins to Thachuk via 333

path from Thachuk to A.Rao:
Thachuk to Hsia via 333
Hsia to Ru.Anderson via 351
Ru.Anderson to Weber via 332
Weber to A.Rao via 421